lib/js/components/DatePickers/DatePicker/DatePicker.vue

```diff
@@ -0,0 +1,291 @@
+		DatePickerBox,
+		DsTile,
+	},
+	props: {
+		triggerType: {
+			type: String as PropType<DatePickerTriggerTypes>,
+			default: DATE_PICKER_TRIGGER_TYPES.TILE,
+		},
+		isInteractive: {
+			type: Boolean,
+			default: true,
+		},
+		placeholder: {
+			type: String,
+			default: 'Wybierz datę',
+		},
+		date: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		label: {
+			type: String,
+			default: '',
+		},
+		isLabelUppercase: {
+			type: Boolean,
+			default: false,
+		},
+		icon: {
+			type: [Object, null],
+			default: ICONS.FA_CALENDAR_DAY,
+			validate: (icon) => icon === null || Object.values(ICONS).includes(toRaw(icon)),
+		},
+		isIconHiddenOnMobile: {
+			type: Boolean,
+			default: false,
+		},
+		calendarPosition: {
+			type: String as PropType<DatePickerCalendarPositions>,
+			default: DATE_PICKER_CALENDAR_POSITIONS.BOTTOM_LEFT,
+		},
+		errorMessage: {
+			type: String,
+			default: '',
+		},
+		state: {
+			type: (String as PropType<DatePickerStates>) || null,
+			default: DATE_PICKER_STATES.DEFAULT,
+		},
+		color: {
+			type: String as PropType<DatePickerColors>,
+			default: DATE_PICKER_COLORS.NEUTRAL,
+		},
+		defaultDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		disableDates: {
+			type: Array,
+			default: () => [],
+		},
+		minDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+		maxDate: {
+			type: [String, Date, Number],
+			default: null,
+		},
+	},
+	emits: { 'update:date': () => true },
+	data() {
+		return {
+			datePicker: null,
+			isOpen: false,
+			DATE_PICKER_STATES: Object.freeze(DATE_PICKER_STATES),
+			DATE_PICKER_TRIGGER_TYPES: Object.freeze(DATE_PICKER_TRIGGER_TYPES),
+		};
+	},
+	computed: {
+		isDatePickerEnabled() {
+			return this.isInteractive && this.state === DATE_PICKER_STATES.DEFAULT;
+		},
+		eyebrowText() {
+			if (!this.date || this.state === DATE_PICKER_STATES.LOADING) {
+				return '';
+			}
+
+			return capitalizeFirstLetter(
+				new Date(this.date).toLocaleDateString(undefined, {
+					dateStyle: undefined,
+					timeStyle: undefined,
+					weekday: 'long',
+				}),
+			);
+		},
+		text() {
+			if (this.state === DATE_PICKER_STATES.LOADING || !this.date?.length) {
+				return this.placeholder;
+			}
+
+			return new Date(this.date).toLocaleDateString(undefined, {
+				dateStyle: 'medium',
+				timeStyle: undefined,
+			});
+		},
+	},
+	async mounted() {
+		if (this.isDatePickerEnabled && !this.datePicker) {
+			if (!flatpickr) {
+				const { default: flatpickrFunction } = await import('flatpickr');
+
+				flatpickr = flatpickrFunction;
+			}
+
+			const { Polish } = await import('flatpickr/dist/l10n/pl.js');
+
+			this.datePicker = flatpickr(this.$refs.flatpickrInput, {
+				locale: Polish,
+				positionElement: this.$el,
+				ignoredFocusElements: [this.$el],
+				appendTo: this.$el,
+				position: FLATPICKR_POSITIONS[this.calendarPosition],
+				defaultDate: this.date,
+				disable: this.disableDates,
+				minDate: this.minDate,
```

Comment:
`minDate` is not changing when props value change, e.g. I have 2 pickers, `start` and `end` date, i set `:min-date="start"`, change `start` date but still i can select date before start

Reply:
what format do you use?

Reply:
returned from @update:date